### PR TITLE
Ejm admin invoice potential revenue

### PR DESCRIPTION
**Pull request by EMielke76**

### CODE REVIEW

**Review by pleonar1 — APPROVED** [review]
I really like what you did to refactor the stuff I did with enabled_items_array and disabled_items_array in the merchant items index view.  I like the refactors in the controller too

**Comment by EMielke76** [comment]
I don't think that was me. I had to commit a merge when I pulled main into my working branch, so that may have been what you were seeing there.

**Comment by pleonar1** [comment]
oh, duh haha my bad

File: app/views/merchant_invoices/show.html.erb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <h1>Invoice <%=@invoice.id%>'s Information</h1>
 <h3>Merchant: <%=@merchant.name%></h3>
-<p>Status: <%=@invoice.status%></p>
-<p>Created On: <%=@invoice.created_at.strftime("%A, %B %d, %Y")%></p>
-<p>Customer: <%=@invoice.customer_name%></p>
+  <p>Status: <%=@invoice.status%></p>
+  <p>Created On: <%=@invoice.created_at.strftime("%A, %B %d, %Y")%></p>
+  <p>Customer: <%=@invoice.customer_name%></p>
```
Comment (LelandCurtis) [inline app/views/merchant_invoices/show.html.erb:5]: why the indent?

File: spec/features/admin/invoices/show_spec.rb [hunk 2]
```diff
@@ -4,12 +4,92 @@
...
+    within("#invoice_#{invoice.invoice_items.first.id}") do
+      expect(page).to have_content("Status: shipped")
+    end
+
+    within("#invoice_#{invoice.invoice_items.second.id}") do
+      expect(page).to have_content("Status: pending")
+    end
+  end
+
+  it 'calculates the potential revenue of the invoice' do
```
Comment (LelandCurtis) [inline spec/features/admin/invoices/show_spec.rb:84]: we should add more test data to check for invoice_items that are not associated with successful transactions.

File: spec/models/invoice_spec.rb [hunk 3]
```diff
@@ -60,5 +60,16 @@
         expect(invoice1.potential_revenue(merchant1)).to eq(35000)
       end
     end
+
```
Comment (LelandCurtis) [inline spec/models/invoice_spec.rb:63]: we should add more test data to check that it doesn't pick up invoice_items that don't have successful transactions. Per User Story 28 potential_revenue description.

**Review by LelandCurtis — APPROVED** [review]
we will need to refactor the potential_revenue methods to not pick up invoice_items that do not have successful transactions associated with them.

File: spec/models/invoice_spec.rb (hunk 3 above)
Comment (EMielke76) [reply spec/models/invoice_spec.rb:63]: Yo, I can definitely work on that

**Pull request merged**